packages/nuxt/test/vite/sourceMaps.test.ts

```diff
@@ -148,146 +151,187 @@ describe('getPluginOptions', () => {
-        nitroConfig.rollupConfig = { output };
-        const previousUserSourceMapSetting = changeRollupSourceMapSettings(nitroConfig, sentryModuleOptions);
-        expect(nitroConfig.rollupConfig?.output?.sourcemap).toBe(expectedSourcemap);
-        expect(previousUserSourceMapSetting).toBe(expectedReturn);
-        expect(nitroConfig.rollupConfig?.output?.sourcemapExcludeSources).toBe(false);
-      });
+    it('should log a warning when Nuxt and Nitro source map settings differ', () => {
+      const nuxt = getNuxtConfig(true);
+      const nitroConfig = getNitroConfig(false);
+
+      validateNitroSourceMapSettings(nuxt, nitroConfig, { debug: true });
+
+      expect(consoleWarnSpy).toHaveBeenCalledWith(
+        "[Sentry] Source map generation settings are conflicting. Sentry uses `sourcemap.server: true`. However, a conflicting setting was discovered (`nitro.sourceMap: false`). This setting was probably explicitly set in your configuration. Sentry won't override this setting but it may affect source maps generation and upload. Without source maps, code snippets on the Sentry Issues page will remain minified.",
+      );
+    });
+
+    it('should set sourcemapExcludeSources to false', () => {
+      const nitroConfig = getNitroConfig(true);
+      validateNitroSourceMapSettings(getNuxtConfig(true), nitroConfig, { debug: true });
+
+      expect(nitroConfig?.rollupConfig?.output?.sourcemapExcludeSources).toBe(false);
     });
+
+    it('should not show console.warn when rollup sourcemap is undefined', () => {
+      const nitroConfig = getNitroConfig(true);
+
+      validateNitroSourceMapSettings(getNuxtConfig(true), nitroConfig, { debug: true });
+
+      expect(consoleWarnSpy).not.toHaveBeenCalled();
+    });
+  });
+});
+
+describe('change Nuxt source map settings', () => {
+  let nuxt: { options: { sourcemap: { client: boolean | 'hidden'; server: boolean | 'hidden' } } };
+  let sentryModuleOptions: SentryNuxtModuleOptions;
+
+  beforeEach(() => {
+    // @ts-expect-error - Nuxt types don't accept `undefined` but we want to test this case
+    nuxt = { options: { sourcemap: { client: undefined } } };
+    sentryModuleOptions = {};
   });
 
-  describe('changeNuxtSourcemapSettings', () => {
-    let nuxt: { options: { sourcemap: { client: boolean | 'hidden'; server: boolean | 'hidden' } } };
-    let sentryModuleOptions: SentryNuxtModuleOptions;
+  it('should handle nuxt.options.sourcemap.client settings', () => {
+    const cases = [
+      { clientSourcemap: false, expectedSourcemap: false, expectedReturn: 'disabled' },
+      { clientSourcemap: 'hidden', expectedSourcemap: 'hidden', expectedReturn: 'enabled' },
+      { clientSourcemap: true, expectedSourcemap: true, expectedReturn: 'enabled' },
+      { clientSourcemap: undefined, expectedSourcemap: 'hidden', expectedReturn: 'unset' },
+    ];
 
-    beforeEach(() => {
+    cases.forEach(({ clientSourcemap, expectedSourcemap, expectedReturn }) => {
```

Comment:
l: We can rewrite this as well as the cases below to `it.each` instead of iterating over the loop. My main concern is seeing what exactly failed which is a bit difficult in the loop (especially when just checking CI logs).